# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.71; am-1.16; lt-2.4.7)
libtoolize -c -f; aclocal; autoconf; automake -a -c
./configure --disable-static --prefix=

# build
make

# install
install -d $DESTDIR/include
install -m 644 include/uv.h $DESTDIR/include
install -d $DESTDIR/include/uv
install -m 644 include/uv/errno.h $DESTDIR/include/uv
install -m 644 include/uv/linux.h $DESTDIR/include/uv
install -m 644 include/uv/threadpool.h $DESTDIR/include/uv
install -m 644 include/uv/unix.h $DESTDIR/include/uv
install -m 644 include/uv/version.h $DESTDIR/include/uv
install -d $DESTDIR/lib
install .libs/libuv.so.1.* $DESTDIR/lib/libuv.so.1
ln -sf libuv.so.1 $DESTDIR/lib/libuv.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libuv.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libuv-build.pc << EOF
Name: libuv
Version: 1.44.2
Description: asynchronous event notification library
Requires: $REQUIRES

devel=\\
/include/uv.h \\
/include/uv/errno.h \\
/include/uv/linux.h \\
/include/uv/threadpool.h \\
/include/uv/unix.h \\
/include/uv/version.h \\
/lib/libuv.so \\
/lib/pkgconfig/libuv.pc

exec=\\
/lib/libuv.so.1
EOF

exit
--- libuv-1.44.2.orig/configure.ac
+++ libuv-1.44.2/configure.ac
@@ -18,6 +18,7 @@
 m4_include([m4/libuv-extra-automake-flags.m4])
 m4_include([m4/as_case.m4])
 m4_include([m4/libuv-check-flags.m4])
+AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects] UV_EXTRA_AUTOMAKE_FLAGS)
 AC_CANONICAL_HOST
 AC_ENABLE_SHARED
@@ -40,8 +41,6 @@
 CC_CHECK_CFLAGS_APPEND([-Wstrict-prototypes])
 # AM_PROG_AR is not available in automake v0.11 but it's essential in v0.12.
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
-# autoconf complains if AC_PROG_LIBTOOL precedes AM_PROG_AR.
-AC_PROG_LIBTOOL
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 LT_INIT
 AX_PTHREAD([
--- libuv-1.44.2.orig/m4/ax_pthread.m4
+++ libuv-1.44.2/m4/ax_pthread.m4
@@ -294,7 +294,7 @@
              # step
              ax_pthread_save_ac_link="$ac_link"
              ax_pthread_sed='s/conftest\.\$ac_ext/conftest.$ac_objext/g'
-             ax_pthread_link_step=`$as_echo "$ac_link" | sed "$ax_pthread_sed"`
+             ax_pthread_link_step=`AS_ECHO(["$ac_link"]) | sed "$ax_pthread_sed"`
              ax_pthread_2step_ac_link="($ac_compile) && (echo ==== >&5) && ($ax_pthread_link_step)"
              ax_pthread_save_CFLAGS="$CFLAGS"
              for ax_pthread_try in '' -Qunused-arguments -Wno-unused-command-line-argument unknown; do
